webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/SecretService.scala

```diff
@@ -0,0 +1,164 @@
+/*
+*************************************************************************************
+* Copyright 2021 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+package com.normation.rudder.web.services
+
+import better.files.Dsl.SymbolicOperations
+import better.files.File
+import com.normation.errors.{IOResult, Inconsistency}
+import net.liftweb.json.JsonAST.JValue
+import net.liftweb.json.JsonDSL._
+import net.liftweb.json.Serialization.write
+import net.liftweb.json.{JString, JsonParser}
+import zio._
+import zio.syntax._
+import java.nio.charset.StandardCharsets
+
+final case class Secret(name: String, value: String)
+
+class SecretVaultService(
```

Comment:
You should define a trait "SecretService" with the abstract functions we need here
add, update, delete, get, getall

Rename the implementing Service to FileSystemSecretService, extending our SecretService , in fact it should even be a "Repository" more than a "Service" 

Service exposing the operation we need at upper level, and calling the repository which will manage specific behafior of the repository (here init of the file ? commiting it too) 

Those function don't need the format as parametern neither does the implementing service